mergify_engine/actions/merge.py

```diff
@@ -37,13 +39,30 @@ class MergeAction(actions.Action):
 
     def __call__(self, installation_id, installation_token, subscription,
                  event_type, data, pull):
+        pull.log.debug("process merge", config=self.config)
 
-        # TODO(sileht): set correct error message if mergeable_state !=
-        # unstable/clean about conflict with the branch protection
-        # TODO(sileht): some error should be report in Github UI
-        # TODO(sileht): Don't blindly update all PRs, but just one by one.
+        # NOTE(sileht): Take care of all branch protection state
+        if pull.g_pull.mergeable_state == "dirty":
+            return None, "Merge conflict needs to be solved"
+        elif pull.g_pull.mergeable_state == "unknown":
+            return ("failure", "Pull request state reported as `unknown` by "
+                    "Github")
+        elif pull.g_pull.mergeable_state == "block":
+            return (None, "Branch protection settings are blocking automatic "
+                    "merging of Mergify")
+        elif (pull.g_pull.mergeable_state == "behind" and
+              not self.config["strict"]):
+            # Strict mode has been enabled in branch protection but not in
+            # mergify
+            return ("failure", "Branch protection setting 'strict' conflicts "
+                    "with Mergify configuration.")
```

Comment:
final dot or no final dot? you never put those.

Reply:
removed